monix-tail/shared/src/main/scala/monix/tail/internal/IterantDropLast.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.tail.internal
+
+import cats.effect.Sync
+import cats.syntax.all._
+import monix.execution.misc.NonFatal
+import monix.tail.Iterant
+import monix.tail.Iterant.{Halt, Last, Next, NextBatch, NextCursor, Suspend}
+import monix.tail.batches.{Batch, BatchCursor}
+
+import scala.collection.mutable
+
+private[tail] object IterantDropLast {
+  /**
+    * Implementation for `Iterant#dropLast`
+    */
+  def apply[F[_], A](source: Iterant[F, A], n: Int)(implicit F: Sync[F]): Iterant[F, A] = {
+
+    def processCursor(toDrop: Int, length: Int, queue: mutable.Queue[A],
+                      ref: NextCursor[F, A]): Iterant[F, A] = {
+      val NextCursor(cursor, rest, stop) = ref
+      val limit = cursor.recommendedBatchSize
+      val buffer = mutable.Buffer[A]()
+
+      var bufferLength = 0
+      var queueLength = length
+      var toProcess = limit
+
+      // protects against infinite cursors
+      while (toProcess > 0 && cursor.hasNext()) {
+        queue.enqueue(cursor.next())
+        toProcess -= 1
+        if (queueLength >= toDrop && bufferLength < limit) {
```

Comment:
I don't think `bufferLength` can ever go over `limit`, since you already guard for `toProcess > 0`.  I might be mistaken, but I think you can get rid of both `bufferLength` and `limit`.

Reply:
I think you're right, about 15 minutes ago I realized I have bugs related to infinite cursors and it became irrelevant but I've forgotten to get rid of it, will do it in the evening, thanks